src/race.rs

```diff
@@ -64,6 +65,21 @@ impl OnceNonZeroUsize {
         }
     }
 
+    /// Sets the contents of this cell to `value`.
+    ///
+    /// Returns `Ok(())` if the cell was empty and `Err(())` if it was
+    /// full.
+    #[cfg(not(target_has_atomic = "ptr"))]
+    #[inline]
+    pub fn set(&self, value: NonZeroUsize) -> Result<(), ()> {
+        if self.inner.load(Ordering::Acquire) == 0 {
+            self.inner.store(value.get(), Ordering::Release);
```

Comment:
Hm, this feels incorrect to me. 

Consider threads A, B, C and the following sequence of events:


```
   A   |   B     |   C
       |         |
load 0 |         |
       | load 0  |
       | store 1 |
       |         | load 1
store 2|         |
       |         | load 2
```

Here, A and B race when initializing the `inner`, so C observes two different values of once cell, which violates the invariant that makes the whole construct a safe abstraction.